controllers/dspipeline_controller.go

```diff
@@ -420,7 +444,7 @@ func (r *DSPAReconciler) GenerateStatus(ctx context.Context, dspa *dspav1alpha1.
 	crReady := r.buildCondition(config.CrReady, dspa, config.MinimumReplicasAvailable)
 	crReady.Type = config.CrReady
 
-	componentConditions := []metav1.Condition{apiServerReady, persistenceAgentReady, scheduledWorkflowReady}
+	componentConditions := []metav1.Condition{databaseAvailable, objStoreAvailable, apiServerReady, persistenceAgentReady, scheduledWorkflowReady}
```

Comment:
the problem outlined in `2.` is likely because the db/objstore `Message` field is not set on failure, so the list of failures is just two empty strings concatted.

Reply:
1. implemented a `disableHealthCheck` override in the CR Spec.

2. Confirmed this was from 2 empty Messages being aggregated together with newline characters only. (the `2` in `|2` is an indentation indicator, and there are 2 newlines following, so the equivalent string being set here was `  \n  \n`.  Adding status messages resolved this.